chia/util/ints.py

```diff
@@ -50,15 +56,21 @@ def stream(self, f):
 
 
 class int512(int):
+    def __new__(cls: Any, value: int):
+        value = int(value)
+        if value >= (2 ** 512) or value <= -(2 ** 512):
+            raise ValueError(f"Value {value} of does not fit into in512")
+        return int.__new__(cls, value)  # type: ignore
+
     # Uses 65 bytes to fit in the sign bit
     @classmethod
     def parse(cls, f: BinaryIO) -> Any:
         read_bytes = f.read(65)
         assert len(read_bytes) == 65
         n = int.from_bytes(read_bytes, "big", signed=True)
-        assert n <= (2 ** 512) - 1 and n >= -(2 ** 512)
+        assert n < (2 ** 512) and n > -(2 ** 512)
         return cls(n)
 
     def stream(self, f):
-        assert self <= (2 ** 512) - 1 and self >= -(2 ** 512)
```

Comment:
this makes the valid range regular, where it's `+(2^512)-1` and `-((2^512)-1)`. This type isn't treated as a normal signed integer. We use 65 bytes to encode it, but then constrain the values as if it was a `uint512` with a sign bit.

This change is mostly unrelated to the other checks being added